core/domain/acl_decorators.py

```diff
@@ -126,3 +167,320 @@ def test_can_view_stats(self, exploration_id, **kwargs):
             raise self.PageNotFoundException
 
     return test_can_view_stats
+
+
+def can_create_exploration(handler):
+    """Decorator to check whether the user can create an exploration."""
+
+    def test_can_create(self, **kwargs):
+        if (
+                (role_services.ACTION_CREATE_EXPLORATION in self.actions) and
+                user_services.has_fully_registered(self.user_id)):
```

Comment:
This has_fully_registered thing -- I'm not sure it should be here. Can this be dealt with at the outset somewhere, so that by the time execution gets to this point, a non-None user_id implies that the user is fully registered? Ditto below.